File: 프로그래머스/0/120907. OX퀴즈/OX퀴즈.js
function solution(quiz) {
    var answer = [];
    for(let i = 0; i< quiz.length; i++){
        var calculateSentence = quiz[i].split(" ")
        var result = Number(calculateSentence[0]);
        for(let j = 1; j< calculateSentence.length; j++ ){
            if(calculateSentence[j] === "+"){
               result += Number(calculateSentence[j+1])
            }else if(calculateSentence[j] === "-"){
               result -= Number(calculateSentence[j+1])    
            }else if(calculateSentence[j] === "="){
                if(result === Number(calculateSentence[j+1])){
                    answer.push('O')
                }else{
                    answer.push('X')
                }
            }
        }
    }
    return answer
}

File: 프로그래머스/0/120907. OX퀴즈/OX퀴즈.java
class Solution {
    public String[] solution(String[] quiz) {
        String[] answer = new String[quiz.length];
        for(int i = 0; i< quiz.length; i++){
            String[] temp = quiz[i].split(" = ");
            String[] compare = temp[0].split(" ");
            int result = Integer.parseInt(temp[1]);
            int compareResult = 0;
            if(compare[1].equals("+")){
              compareResult = Integer.parseInt(compare[0]) + Integer.parseInt(compare[2]);
            }else{
              compareResult = Integer.parseInt(compare[0]) - Integer.parseInt(compare[2]);
            }
            if(compareResult == result){
                answer[i] = "O";
            }else{
                answer[i] = "X"; 
            }
        }
 
        return answer;
    }
}

